### PR TITLE
Added API for row details (now with tests!)

### PR DESCRIPTION
**Pull request by tepi**
[Review on Reviewable](https://reviewable.io/reviews/vaadin/components/98)

### CODE REVIEW

File: vaadin-components-gwt/src/main/webapp/vaadin-grid/vaadin-grid.html [hunk 1]
```diff
@@ -970,6 +970,40 @@
...
+      },
+
+      /**
+       * The row detail generator that is used for generating detail content for rows.
+       *
+       * Example:
+       * ```
+       * grid.rowDetails.detailsGenerator = function(rowIndex) {
+       *  var detail = document.createElement('div');
+       *  detail.innerText = 'Row detail content for row ' + rowIndex;
```
Comment (jounik) [inline vaadin-components-gwt/src/main/webapp/vaadin-grid/vaadin-grid.html]: `element.innerText` is only supported in some browsers, it’s not a standard property. Use `element.textContent` instead.

**Comment by jounik** [comment]
Apart from `element.innerText`, LGTM.

**Comment by tepi** [comment]
Changed innerText to textContent.

**Comment by tomivirkki** [comment]
<sup>**[vaadin-components-gwt/src/main/webapp/vaadin-grid/vaadin-grid.html, line 992 \[r2\]](https://reviewable.io:443/reviews/vaadin/components/98#-JxESOxaMl_WC5xJFcKy)** ([raw file](https://github.com/vaadin/components/blob/d072e39ae47234095a1f54c018c73b2c855f6172/vaadin-components-gwt/src/main/webapp/vaadin-grid/vaadin-grid.html#L992)):</sup>
This is not a parameter for the grid.rowDetailGenerator api though so it shouldn't be here.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/components/98)

**Comment by tomivirkki** [comment]
<sup>**[vaadin-components-gwt/src/main/java/com/vaadin/components/grid/GridComponent.java, line 600 \[r2\]](https://reviewable.io:443/reviews/vaadin/components/98#-JxESpXRc18DRxiWwTnR)** ([raw file](https://github.com/vaadin/components/blob/d072e39ae47234095a1f54c018c73b2c855f6172/vaadin-components-gwt/src/main/java/com/vaadin/components/grid/GridComponent.java#L600)):</sup>
Primitive parameters coming from JS should be validated

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/components/98)

**Comment by tomivirkki** [comment]
Reviewed 1 of 2 files at r2, 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks pending.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/components/98)

**Comment by tomivirkki** [comment]
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks pending.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/components/98)

**Pull request merged**